src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -602,14 +1443,34 @@ ErrorHintRequired(const char *errorHint, Query *queryTree)
 
 
 /*
- * ErrorIfSubqueryNotSupported checks that we can perform distributed planning for
- * the given subquery.
+ * DeferErrorIfSubqueryNotSupported checks that we can perform distributed planning for
+ * the given subquery. If not, a deferred error is returned. The function recursively
+ * does this check to all lower levels of the subquery.
  */
-static void
-ErrorIfSubqueryNotSupported(Query *subqueryTree)
+static DeferredErrorMessage *
+DeferErrorIfUnsupportedSubqueryRepartition(Query *subqueryTree)
 {
 	char *errorDetail = NULL;
 	bool preconditionsSatisfied = true;
+	List *joinTreeTableIndexList = NIL;
+	int rangeTableIndex = 0;
+	uint32 joiningRangeTableCount = 0;
+	RangeTblEntry *rangeTableEntry = NULL;
+	Query *innerSubquery = NULL;
+
+	/*
+	 * Extract all range table indexes from the join tree. Note that sub-queries
+	 * that get pulled up by PostgreSQL don't appear in this join tree.
+	 */
+	ExtractRangeTableIndexWalker((Node *) subqueryTree->jointree,
+								 &joinTreeTableIndexList);
+
+	joiningRangeTableCount = list_length(joinTreeTableIndexList);
+	if (joiningRangeTableCount > 1)
```

Comment:
Sorry, we could hit this line. This is basically necessary to generate the error message for that case.